File: introducao/portas.c
#include <stdio.h>

#define MAX_TAM 100
#define TRUE 1
#define FALSE 0

int main()
{
    char predio[MAX_TAM];
    int i, A, N, x, numero;
    while (TRUE)
    {
        scanf("%d %d", &A, &N);
        if (A == 0 && N == 0) // condicao de parada
            break;
        for (i = 0; i < A; i++) // inicializa o predio
            predio[i] = FALSE;
        for (i = 0; i < N; i++) // leitura dos numeros
        {
            scanf("%d", &numero);
            for (x = numero - 1; x < A; x += numero)
                predio[x] = !predio[x];
        }
        for (i = 0; i < A; i++) // impressao do predio
            printf("%c", predio[i] ? 'O' : 'C');
        printf("\n");
    }
    return 0;
}

File: introducao/portas.py
MAX_TAM = 100
TRUE = True
FALSE = False

while TRUE:
    A, N = map(int, input().split())
    # Condição de parada
    if A == 0 and N == 0:
        break
    
    # Inicializa o prédio
    predio = [FALSE] * A
    
    # Leitura dos números
    for _ in range(N):
        numero = int(input())
        for x in range(numero - 1, A, numero):
            predio[x] = not predio[x]
    
    # Impressão do prédio
    print(''.join(['O' if p else 'C' for p in predio]))
